packages/ui/src/components/Pagination/SPagination.vue

```diff
@@ -0,0 +1,377 @@
+<script setup lang="ts">
+import {
+  IconArrowsChevronRightXs24,
+  IconArrowsChevronLeftXs24,
+  IconChevronsLeft16,
+  IconChevronsRight16,
+} from '@/components/icons'
+import { not } from '@vueuse/core'
+import { SDropdown } from '@/components'
+
+const PAGINATION_MAX_PAGES_SELECTABLE = 7
+const PAGINATION_JUMP_SIZE = PAGINATION_MAX_PAGES_SELECTABLE - 2
+const PAGINATION_BREAKPOINT_WIDTH = 480
+
+type PaginationJumpValues = number | 'jumpNext' | 'jumpPrev'
+
+const props = withDefaults(
+  defineProps<{
+    /**
+     * total item count
+     * */
+    total?: number
+    /**
+     * item count of each page, two-way bound
+     * */
+    pageSize?: number | null
+    /**
+     * current page number, two-way bound
+     * */
+    currentPage?: number
+    /**
+     * options of item count per page
+     * */
+    pageSizes?: number[]
+    /**
+     * text shown near sizes dropdown
+     * */
+    sizesLabel?: string
+  }>(),
+  {
+    total: 0,
+    pageSize: null,
+    currentPage: 1,
+    pageSizes: () => [10, 50, 100],
+    sizesLabel: 'Rows per page',
+  },
+)
+/* eslint-disable @typescript-eslint/unified-signatures */
+const emit = defineEmits<{
+  (event: 'size-change', value: number): void
+  (event: 'current-change', value: number): void
```

Comment:
What are these events for? We already have `update:*` for them. Is it only for compatibility reasons? If so, I don't think that it worth it to keep such things only for compatibility reasons.